tests/scripts/basic-build-test.sh

```diff
@@ -62,6 +62,11 @@ fi
 : ${SEED:=1}
 export SEED
 
+# if MAKEFLAGS is not set add the -j option to speed up invocations of make
+if [ -z "${MAKEFLAGS+set}" ]; then
+    export MAKEFLAGS="-j"
```

Comment:
Re. the `-j` option - according to `make` man-page: *"If the `-j` option is followed by an integer, this is the number of recipes to execute at once; this is called the number of job slots. **If there is nothing looking like an integer after the `-j` option, there is no limit on the number of job slots.**"* So my suspicion here is that bare `-j` option used in nightlies (PR #4773) can easily overload the machine when used in too many tasks. So I would suggest to use here something like `-j$(nproc)`.

Reply:
Indeed, but I'm not sure this matters.

I kept `-j` because that's what was there before and it's what's in `all.sh`, and it's also what some people want on their own machine. In our Jenkins scripts, we override `MAKEFLAGS`, but I just checked and we only do that for the `all.sh` jobs, not for `basic-build-test.sh`.

The reason we overrode `MAKEFLAGS` is that we had load spikes that caused individual test cases to time out because a process spent several seconds without being responsive. Moving away from parallel builds seems to have helped with this (but we changed other things around the same time so it's hard to be sure how much the parallel builds to the problem).

Parallel builds don't have much impact on the total time: the dominant part of the long jobs is TLS testing, not building. So the value of `MAKEFLAGS` has little impact on the time or resources consumed by a full CI run.